pkg/kubelet/kubelet_pods.go

```diff
@@ -338,6 +342,12 @@ func nodeHostsFileContent(hostsFilePath string, hostAliases []v1.HostAlias) ([]b
 	if err != nil {
 		return nil, err
 	}
+	// If we are running in a DIND environment, the /etc/hosts file already has the
+	// file header, so we should strip it out since the file header is used to signal
+	// that the container's /etc/hosts is managed.
+	if bytes.HasPrefix([]byte(hostsFileContent), []byte(managedFileHeader)) {
```

Comment:
A quick question:

Why don't use [strings.Index(str, substr)](https://golang.org/pkg/strings/#Index)?

Reply:
sorry i believe histsFileContent is already []byte, let me fix that